pandas-stubs/core/frame.pyi

```diff
@@ -164,9 +167,33 @@ class DataFrame:
         downcast: Dict = ...,
     ) -> None: ...
     @overload
-    def groupby(self, by: List[_str]) -> DataFrameGroupBy: ...
+    def groupby(
+        self,
+        by: Union[
+            _str,
+            Tuple[_str, ...],
+            List[_str],
+            List[Tuple[_str, _str]],
+            List[Tuple[_str, _str, _str]],
+        ],
+        level: Union[int, _str] = ...,
+        as_index: bool = ...,
+        sort: bool = ...,
+        group_keys: bool = ...,
+        squeeze: bool = ...,
+        observed: bool = ...,
+    ) -> DataFrameGroupBy: ...
     @overload
-    def groupby(self, by: _str) -> SeriesGroupBy: ...
+    def groupby(
+        self,
+        by: Union[Series[_str], Dict[_str, _str], Callable],
+        axis: _AxisType = ...,
+        level: Union[int, _str] = ...,
+        sort: bool = ...,
+        group_keys: bool = ...,
+        squeeze: bool = ...,
+        observed: bool = ...,
+    ) -> DataFrameGroupBy: ...
```

Comment:
wait, these two overloads now return the same thing. what happened to the overload that returns `SeriesGroupBy`?

Reply:
Yeah that's right, the `groupby()` never returned a `SeriesGroupBy` as long as I tested it, and so says the [documentation](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.groupby.html). The only way to get a `SeriesGroupBy` was to extract a column from a `DataFrameGroupBy`. If you succeed in getting a `SeriesGroupBy` from the `groupby()` method, could you please provide the code?

The two versions of `groupby` you are seeing now differ by the axis.
Single index (which may be multiindex) and list of indices work with the default `axis=0`, and the `as_index` argument is allowed, like this:
```
iris.groupby('species', as_index=False)
```
The second version work with mappings, that may be `dict`, `Series` or `function`, it requires `axis=1` and does not allow `as_index=False`. The example (dropping the species column, which is not numeric, may be required for aggregating):
```
iris.groupby({
    'sepal_length': 'length',
    'petal_length': 'length',
}, axis=1)
```
Both versions return `DataFrameGroupBy`.

Reply:
I see! Thanks for investigating this!